sycl/source/detail/scheduler/circular_buffer_extended.cpp

```diff
@@ -0,0 +1,133 @@
+//==-- circular_buffer_extended.hpp - Circular buffer with host accessor ---==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <detail/scheduler/circular_buffer_extended.hpp>
+#include <detail/scheduler/scheduler.hpp>
+
+#include <algorithm>
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace detail {
+
+static inline bool doOverlap(const Requirement *LHS, const Requirement *RHS) {
```

Comment:
Looks like this duplicates the function defined in `graph_builder.cpp`

Reply:
Done